src/main/kotlin/se/uulm/snowballr/backend/env/EnvService.kt

```diff
@@ -54,6 +60,12 @@ class EnvService : IEnvService {
             ?: throw EnvVariableNotFoundException("Invalid boolean value for key '$key': '$value'")
     }
 
+    override fun getRequiredOrDefault(key: String, default: String?): String {
+        return default?.let { getOrDefault(key, it) }
+            ?: dotenv[key]
+            ?: throw EnvVariableNotFoundException(key)
+    }
```

Comment:
I like this, but it's a bit difficult to read. What about splitting this expression into two parts?

```suggestion
    override fun getRequiredOrDefault(key: String, default: String?): String {
        if (default != null) {
            return getOrDefault(key, default)
        }

        return dotenv[key] ?: throw EnvVariableNotFoundException(key)
    }
```